### ISSUE TITLE
Исследовательская часть:Описание предметной области

### ISSUE THREAD

**Issue by alex309706** [alex309706/Confectionery#7]
Проработать досконально сущности и взаимосвязи между ними.
Оформить исследовательскую часть по ГОСТУ

**Comment by wldmr-key** [comment]
1) про гит. Обрати внимание -  у тебя здесь есть файл с описанием и в документации есть. На какой мне смотреть? Сам не запутаешься? не стоит в задачи загружать файлы (во всяком случае, многократного использования). Для этого и придумана система версий
2) собственно про описание
Общие сведения: недостаточны. Лирика до предпоследнего абзаца приемлема, но не особенно нужна. Зачем тебе вообще сырье для сайта? - это вопрос, а не порицание: из дальнейшего изложения не видно в каком процессе это используется. Вообще описания процессов нет и это осложняет понимание того, что будет сделано.          Как происходит, например, заказ?
Перечень реализуемых функций очень неконкретен и применим к чему угодно
Из-за недостаточной проработанности этих двух подразделов концептуальная схема неправильна не только по форме, но и по существу. В среду на лекции я разберу еще раз принципы составления er-диаграмм
задача автоматизации в этом контексте - это уже, действительно автоматизация (воплощение в программном коде) выбранных тобой процессов

**Comment by alex309706** [comment]
1)нет,не запутаюсь,ведь у меня всё это на компе..хорошо,буду в документацию скидывать.. 
2)скинул еще версию описания предметной области..Заказ осуществляется следующим образом: понравившиеся изделия заносятся в список (корзину), выбирается количество изделий, сумма автоматически подсчитывается, при нажатии на кнопку «заказать» выскакивает ссылка на оплату заказа...

**Comment by alex309706** [comment]
[Предметная область.docx](https://github.com/alex309706/Confectionery/files/2552239/default.docx)
Добрый день! Исправил концептуальную схему .Но, кажется, связей мало. Есть еще вопрос: реклама фабрики и информация о ней(вакансии,награды и пр.)..для этого разве не нужны сущности? Например, так и назвать, "информация".. или в моём случае нужно только то, что связано с заказами?

**Comment by wldmr-key** [comment]
Связей не то чтобы мало, но они неправильно нарисованы. На словах ты говоришь все правильно, а на схеме ошибаешься. Кондитерское изделие у тебя висит в вакууме, а на словах оно входит в заказ. На словах у тебя пользователь связан только с заказом, а на картинке еще появляется связь с информацией о заказе. Нужно исправлять картинку
Что касается статических страниц сайта, то, конечно, ты можешь хранить их в базе, но есть ли в этом смысл? Если есть, то, пожалуйста...

**Comment by alex309706** [comment]
[Предметная область.docx](https://github.com/alex309706/Confectionery/files/2554869/default.docx)
Переделал связи..Добавил сценарии использования

**Comment by wldmr-key** [comment]
Ох, Александр... Графические схемы, видимо, не самая твоя сильная сторона...

Каждое кондитерское изделие входит ровно в один заказ? Точно?

А то, что ты называешь сценариями использования таковыми не является. Глянь в википедии что-ли как они описываются и рисуются

**Comment by alex309706** [comment]
Исправил.В заказ может входить несколько кондитерских изделий. и любое кондитерское изделие может быть заказано.
Да...видимо, проблемы с этим есть..
Сценарий больше похож на алгоритм..Если это не то,что нужно, то подозреваю,что требуется более конкретно прописывать шаги..так? ..и да, схемку для сценария тоже нужно будет сделать,правильно?
[Предметная область.docx](https://github.com/alex309706/Confectionery/files/2556329/default.docx)

**Comment by wldmr-key** [comment]
нет, неправильно - в заказ, действительно, может входить несколько изделий, но в разных количествах. Это у тебя отражается в сущности "информация о заказе" или, другими словами, в позиции заказа - там **только один тип** изделий и его количество. Так что стрелка неправильная - нужно один ко многим
Алгоритм - это тоже неплохо. Просто желательно находиться в рамках общепринятых формализмов, иначе тебя плохо поймут

**Comment by alex309706** [comment]
[Предметная область.docx](https://github.com/alex309706/Confectionery/files/2602697/default.docx)
Исправил схему и сценарии.
